application/comit_node/src/settings/mod.rs

```diff
@@ -110,4 +208,35 @@ mod tests {
             .is_equal_to(&Duration::from_secs(20));
     }
 
+    #[test]
+    fn config_folder_does_not_exist_will_create_folder_and_config_file() {
+        let tmp_dir = env::temp_dir();
+        let config_path = Path::join(&tmp_dir, "i_am_invincible");
+        let config_file = "default.toml";
+
+        delete_tmp_files(&config_path, config_file);
+
+        let config_file_incl_path = config_path.clone().join(config_file.clone());
+
+        let default_settings = ComitNodeSettings::default();
+
+        let default_settings = default_settings.write_to(config_file_incl_path.clone());
+        let settings = ComitNodeSettings::read(config_file_incl_path.clone());
+
+        delete_tmp_files(&config_path, &config_file);
+
+        assert_that(&default_settings).is_ok();
+        assert_that(&settings).is_ok();
+        assert_that(&default_settings.unwrap()).is_equal_to(&settings.unwrap());
```

Comment:
:wrench: You verified that their are `Ok` already, no need for the unwrap. You could just compare the results.
```suggestion
        assert_that(&default_settings).is_equal_to(&settings);
```

Reply:
Unfortunately not:
```
    = note: the method `is_equal_to` exists but the following trait bounds were not satisfied:
            `std::result::Result<settings::ComitNodeSettings, config::error::ConfigError> : std::cmp::PartialEq`
```